### PR TITLE
[AIRFLOW-4565] instrument celery executor

### PR DESCRIPTION
**Pull request by youngyjd**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW-4565) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4565
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:
Extract out a new method called trigger_tasks from hearbeat method, so CeleryExecutor can get instrumented as well.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: airflow/executors/celery_executor.py [hunk 1]
```diff
@@ -182,10 +183,17 @@ def heartbeat(self):
         else:
             open_slots = self.parallelism - len(self.running)
 
-        self.log.debug("%s running task instances", len(self.running))
-        self.log.debug("%s in queue", len(self.queued_tasks))
+        num_running_tasks = len(self.running)
```
Comment (milton0825) [inline airflow/executors/celery_executor.py]: Looks like we have similar code in:
https://github.com/apache/airflow/blob/a9eff7a7e4367414040c6fb1cf17f30bcaf0dd5c/airflow/executors/base_executor.py#L116-L130

Can we consolidate the two?

Comment (youngyjd) [reply airflow/executors/celery_executor.py]: @milton0825 Does current change look better to you?

File: airflow/executors/base_executor.py [hunk 2]
```diff
@@ -129,6 +129,19 @@ def heartbeat(self):
...
 
+        self.trigger_tasks(open_slots)
+
+        # Calling child class sync method
+        self.log.debug("Calling the %s sync method", self.__class__)
+        self.sync()
+
+    def trigger_tasks(self, open_slots):
+        """
+        Trigger tasks. This function is overwritten in CeleryExecutor
```
Comment (milton0825) [inline airflow/executors/base_executor.py]: I think it is fine that you don't mention `This function is overwritten in CeleryExecutor`

**Review by milton0825 — APPROVED** [review]
Looks good~

**Comment by youngyjd** [comment]
@feng-tao

**Comment by feng-tao** [comment]
is there any existing tests for the heartbeat method ? And given you create the trigger task method, do you think it would be easy to add some tests for baseExtractor and celeryExtractor?

**Comment by youngyjd** [comment]
@feng-tao Added tests for all executors to ensure that executor stats are gauged

**Review by feng-tao — APPROVED** [review]

**Pull request merged**